store/batch.go

```diff
@@ -76,7 +76,11 @@ func (b *batch[H]) getByHeight(height uint64) H {
 		return zero
 	}
 
-	return b.headers[height-base-1]
+	h := b.headers[height-base-1]
+	if h.Height() == height {
```

Comment:
Well, sometimes `b.headers` doesn't mean it's a contiguous header chain. We need to check that the estimated header is the right one.